receiver/elasticsearchreceiver/metadata.yaml

```diff
@@ -137,6 +140,13 @@ attributes:
     enum:
       - hit
       - miss
+  index_aggregation_type:
+    value: type
+    description: Type of aggregation for index statistics
+    enum:
+      - primaries
```

Comment:
What's the rationale for making this plural?

Reply:
JSON data sent by elasticsearch contains two keys inside an index data - `primaries` and `total`, where `primaries` are the values only for primary shards. 
To be honest, I don’t have any idea how to call it accurately. "Primary" is ambiguous - it can also mislead into thinking that it is some kind of "priority" or "the best" aggregation. Does `primary_shards` sound good?

Reply:
I'm also not sure what names would be accurate here. 

What is comes down to for me is that users should be able to figure out what these are referring to, but for whatever reason I'm not able to find documentation that directly describes what these attribute values mean. 

The names themselves may not be super important here, but we should provide a link to some kind of documentation that describes the meaning. This can be done in the `extended_description` field.

Reply:
I couldn't find the `extended_description` field in the docs for `mdatagen`: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/metric-metadata.yaml
There is only `extended_documentation`, but it's related to metrics, not attributes - and this attribute is going to apply to many metrics, so I don't think it's a good idea to write that here.

I decided to change the description, enum variant and short name a bit.